### PR TITLE
NFC: Removed some uses of terms "send" and "receive" in the code, when the context does not make it clear where it's sending to or receiving from. 

### PR DESCRIPTION
**Pull request by mhong**
Instead we are favoring phrases with more explicit context such as "host to accel
transfer" and "send to accelerator/TF".

### CODE REVIEW

**Comment by mhong** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/TFPartition.cpp [hunk 1]
```diff
@@ -3490,10 +3490,6 @@ bool PartitionCloner::finalizeOriginal() {
   // bb arg -> branch argument mappings.
   SmallPtrSet<SILBasicBlock *, 4> argsRemovedBlocks;
 
-  // We remove arguments with a two-pass approach, first dropping the insts that
-  // feed them, then removing the arg (potentially inserting a receive).
-  SmallVector<SILArgument *, 4> argsToRemove;
```
Comment (rxwei) [inline lib/SILOptimizer/Mandatory/TFPartition.cpp]: is this just removing dead code?

Comment (mhong) [reply lib/SILOptimizer/Mandatory/TFPartition.cpp]: Yes this is dead code.

**Review by rxwei — APPROVED** [review]
LGTM. I always found "send"/"receive" (without specifying which direction) in TFPartition confusing as hell. This is a great readability improvement.

File: lib/SILOptimizer/Mandatory/TFPartition.cpp [hunk 2]
```diff
@@ -115,11 +115,11 @@ enum class PartitioningClass {
 
   /// This is an explicit send to the accelerator, which is sugared as
   /// '.toAccelerator()'.
-  ExplicitSend,
+  ExplicitToAccel,
 
   /// This is an explicit receive from the accelerator, which is sugared as
   /// '.toHost()'.
-  ExplicitReceive,
+  ExplicitToHost,
```
Comment (bgogul) [inline lib/SILOptimizer/Mandatory/TFPartition.cpp]: Please correct me if I am wrong. ExplicitToHost would result in a send in the accelerator code and a receive in the host code. Right?

Comment (mhong) [reply lib/SILOptimizer/Mandatory/TFPartition.cpp]: Yes

**Review by lattner — APPROVED** [review]
LGTM!

**Pull request merged**